rasa/core/nlg/interpolator.py

```diff
@@ -6,7 +6,7 @@
 
 
 def interpolate_text(template: Text, values: Dict[Text, Text]) -> Text:
```

Comment:
I think the rename is a bit tricky in this module. 
* we are taking `Domain.responses` and interpolating placeholders for them -> speaks for a rename to `responses`
* A string with placeholders which are interpolated, is a `template` -> speaks for keeping it as it is

What's your take on it?

Reply:
Okay, now that you have explained the code to me, then I'd probably vote for `response`. `template` originally meant `response` and it could be a bit confusing to keep this terminology for this specific use case. `responses` with placeholders are still `responses`. 😅 What do you think?

Reply:
So in this case, we will swap everything that says `template_name` to `action_name`. And then just get rid of `template` altogether.

Reply:
Okay, below I also found another name for things like `utter_greet`. It is called `utter_action`. This actually makes quite a lot of sense. Should we pick one of the 2, `action_name` versus `utter_action`

Reply:
`utter_action` sounds great!